simulator/infra/config/t/unit_test.cpp

```diff
@@ -0,0 +1,242 @@
+/*
+* This is an open source non-commercial project. Dear PVS-Studio, please check it.
+* PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+*/
+/**
+ * Unit tests for configuration parser
+ * @author Denis Los
+ */
+
+
+// generic C
+#include <cstdlib>
+#include <cassert>
+
+
+// Google test library
+#include <gtest/gtest.h>
+
+
+#include "../config.h"
+
+
+namespace config {
+static RequiredValue<std::string> binary_filename = { "binary,b", "input binary file"};
```

Comment:
It's better to be a little more abstract here and use some dummy names like "mandatory_string_value" or "mandatory_int_value".